pipelines/imputation/scientificValidation/ReshapeReferencePanel.wdl

```diff
@@ -4,21 +4,508 @@ version 1.0
+                    start = start_chunk_second,
+                    end = end_chunk_second,
+                    chrom = contig,
+                    basename = chunk_second_basename,
+                    gatk_docker = gatk_docker
+            }
+        }
+    }
+
+    Array[Array[File]] vcfs_to_be_merged_and_gathered = transpose(GenerateChunkSecond.output_vcf)
+
+    scatter (i in range(length(vcfs_to_be_merged_and_gathered))) {
+        call MergeVcfsWithCutPaste {
+            input:
+                vcfs = vcfs_to_be_merged_and_gathered[i],
+                basename = "merge_samples_from_chunk_" + i
+        }
+
+        call CreateVcfIndex as CreateVcfIndexCutPasteMerge {
+            input:
+                vcf_input = MergeVcfsWithCutPaste.output_vcf
+        }
+    }
+
+    call GatherVcfs as GatherVcfsSecond {
+        input:
+            input_vcfs = CreateVcfIndexCutPasteMerge.output_vcf,
+            output_vcf_name = output_base_name + ".reshaped." + contig + ".vcf.gz"
+    }
+
+    output {
+        File output_vcf = GatherVcfsSecond.output_vcf
+        File output_vcf_index = GatherVcfsSecond.output_vcf_index
+    }
+}
+
+task ChunkSampleNames {
+    input {
+        File vcf
+        Int sample_chunk_size
+
+        Int disk_size_gb = ceil(size(vcf, "GiB")) + 10
+        String bcftools_docker = "us.gcr.io/broad-dsde-methods/gatk-sv/denovo:2025-02-11-v1.0.2-hotfix-22bf77e0"
+        Int cpu = 1
+        Int memory_mb = 4000
+    }
+    command {
+        set -e -o pipefail
+
+        export GCS_OAUTH_TOKEN=$(gcloud auth application-default print-access-token)
+
+        # query the sample names from the VCF and chunk by sample_chunk_size
+        bcftools head ~{vcf} > vcf_header.txt
+        bcftools query -l vcf_header.txt > sample_names.txt
+        split -l ~{sample_chunk_size} sample_names.txt sample_chunks
+
+        # store number of samples in a file to be used in downstream scattering/tasks
+        wc -l sample_names.txt | cut -d' ' -f1 > sample_count.txt
+    }
+
+    runtime {
+        docker: bcftools_docker
+        disks: "local-disk ${disk_size_gb} HDD"
+        memory: "${memory_mb} MiB"
+        cpu: cpu
+    }
+
+    parameter_meta {
+        vcf: {
+            description: "vcf",
+            localization_optional: true
+        }
+    }
+
+    output {
+        Array[File] sample_names = glob("sample_chunks*")
+        Int sample_count = read_int("sample_count.txt")
+    }
+}
+
+task SelectSamplesWithCut {
+    input {
+        File vcf
+
+        Int cut_start_field
+        Int cut_end_field
+        String basename
+
+        Int disk_size_gb = ceil(1.5 * size(vcf, "GiB")) + 10
+        String bcftools_docker = "us.gcr.io/broad-gotc-prod/imputation-bcf-vcf:1.0.7-1.10.2-0.1.16-1669908889"
+        Int cpu = 2
+        Int memory_mb = 6000
+    }
+
+    command <<<
+        set -euo pipefail
+
+        mkfifo fifo_bgzip
```

Comment:
I feel like this should be fifo_decompressed lol

Reply:
`bgzip` is just the name of the tool that is outputting to that fifo, just like `cut` is outputting to `fifo_cut`